### PR TITLE
Enable MacOS Catalyst support, missing permissions and tweak webview options

### PR DESCRIPTION
**Pull request by khmyznikov**
MacOS fixes: window header calculation, app icon.
Storyboard: remove redundant backgrounds, fix icon/progress bar alignment.

### CODE REVIEW

**Comment by khmyznikov** [comment]
Added missing permissions according to issue #18
Navigation gestures allowed according to issue #20

**Comment by JudahGabriel** [comment]
Gleb, you are amazing.

Just got back from a quick vacation week and you've already implemented some great things.

Question:

You added camera, location, and microphone permissions. Will the user need to allocate those permissions when [creating their Bundle ID](https://github.com/pwa-builder/pwabuilder-ios/blob/main/submit-to-app-store.md#2-create-your-bundle-id)? If so, we'll need to update our docs to include those.

**Comment by khmyznikov** [comment]
> Just got back from a quick vacation week and you've already implemented some great things.

Glad to try!

> You added camera, location, and microphone permissions. Will the user need to allocate those permissions when [creating their Bundle ID](https://github.com/pwa-builder/pwabuilder-ios/blob/main/submit-to-app-store.md#2-create-your-bundle-id)? If so, we'll need to update our docs to include those.

I don't think so, looks like it doesn't count as [capabilities](https://developer.apple.com/documentation/xcode/adding-capabilities-to-your-app). It works like a passthrough from WebView to prevent unexpected requests, and technically I have added Descriptions, not Permissions.

**Comment by khmyznikov** [comment]
Display web view full-screen issue #17

**Comment by JudahGabriel** [comment]
Thanks. I added a question in the code review, see above.

**Review by JudahGabriel — CHANGES_REQUESTED** [review]

File: Microsoft.PWABuilder.IOS.Web/Resources/ios-project-src/pwa-shell/Base.lproj/Main.storyboard [hunk 1]
```diff
@@ -14,40 +14,38 @@
...
+                                                <color key="tintColor" systemColor="secondarySystemFillColor"/>
+                                            </imageView>
                                         </subviews>
                                         <viewLayoutGuide key="safeArea" id="a0B-Gx-0cO"/>
-                                        <color key="backgroundColor" red="0.0" green="0.0" blue="1" alpha="1" colorSpace="custom" customColorSpace="sRGB"/>
+                                        <rect key="contentStretch" x="0.0" y="0.0" width="0.0" height="0.0"/>
                                     </view>
                                 </subviews>
-                                <color key="backgroundColor" red="0.0" green="1" blue="0.0" alpha="1" colorSpace="custom" customColorSpace="sRGB"/>
+                                <color key="backgroundColor" red="0.22232502700000001" green="0.4557446241" blue="0.7985594869" alpha="1" colorSpace="custom" customColorSpace="displayP3"/>
```
[the remark this answers is not in the document]
Comment (khmyznikov) [reply Microsoft.PWABuilder.IOS.Web/Resources/ios-project-src/pwa-shell/Base.lproj/Main.storyboard:48]: Nothing special, maybe this color was too bright for me and I’m muted it. You can override it back as you need.

**Review by JudahGabriel — APPROVED** [review]

**Comment by JudahGabriel** [comment]
Thanks Gleb! I've merged this in. I'll test it with the generation service, and if all looks good, publish to prod.

**Comment by JudahGabriel** [comment]
Hey @khmyznikov 

Testing your changes out in the template, I see this:

image

Clearly not what we're intending here. Want me to fix it? I just don't want to lose any other changes you did.

**Comment by JudahGabriel** [comment]
In essence, your PR changes the Loading View to be a square in the center of the screen:

image

PWABuilder is currently using Loading View as the splash screen, however. 

So, we have 2 ways of fixing this:

- Make Loading View take up all the space
- Add another element behind Loading View, e.g. Splash View, that takes up all the space and acts as the splash screen background.

Thoughts?

**Comment by khmyznikov** [comment]
@JudahGabriel I think we already discussed something related to this. Looks like it's because I have deleted some of the background colors you inserted. If I remember right, the loading view doesn't need to be stretched to full screen, just paint another View to the same color. I can try to look at it in a few days if you don't find a solution earlier.

**Comment by JudahGabriel** [comment]
Understood. I'll update the code accordingly, leaving the loading view unstretched. Thanks!

**Comment by JudahGabriel** [comment]
Hey Gleb. Yeah, I added a new UIView to Main.storyboard that acts as the background splash color. See https://github.com/pwa-builder/pwabuilder-ios/commit/95685e3ef393f9f1af640464f42a1faa3d853e05#diff-33199783740aa45ff5fbc93443437e903caa7eef0fffd6802829bbbacf9b9580L24-L53

I also updated ViewController.swift to set the top frame of the new UIView to be just underneath the status bar. See https://github.com/pwa-builder/pwabuilder-ios/commit/95685e3ef393f9f1af640464f42a1faa3d853e05#diff-6b2e29fc0cadcddfe086b0a4606f20d23891dde4ca6267a500b3cdef4f484157R85-R88

Publishing this to production today, finally. :-)

**Pull request merged**